q2_phylogeny/_fasttree_with_mafft.py

```diff
@@ -0,0 +1,24 @@
+# ----------------------------------------------------------------------------
+# Copyright (c) 2016-2018, QIIME 2 development team.
+#
+# Distributed under the terms of the Modified BSD License.
+#
+# The full license is in the file LICENSE, distributed with this software.
+# ----------------------------------------------------------------------------
+
+
+def fasttree_with_mafft(ctx, sequences, n_threads=1, max_gap_frequency=1.0,
+                        min_conservation=0.40):
+    mafft = ctx.get_action('alignment', 'mafft')
+    mask = ctx.get_action('alignment', 'mask')
+    fasttree = ctx.get_action('phylogeny', 'fasttree')
```

Comment:
I am wondering if we can generalize this a bit more? That is, add `raxml` (and `iqtree` when it is merged) to the phylogeny options?  Also, not sure if there will be other alignment options added in the future, in which case we can generalize that part too. Unless the idea is to keep things simple for now to gauge its use? Thoughts?

Reply:
I was wondering about that too! One problem though --- pipelines aren't able to dynamically populate their parameters, so we would need to expose all of the params for `fasttree`, `raxml`, and `iqtree`, which sounds pretty gnarly, and really error prone. @ebolyen - any chance we can tackle this in the future?

Reply:
Ahh okay. I've not spent much time reviewing how pipelines can be structured. But I can see how things can get gnarly. 😸

Reply:
this was also on my mind! Otherwise things will get really gnarly with all the different alignment/tree building combinations if we want to make pipelines for different combos.

One possibility: just expose (and rename) the most important parameters that are generalized across methods (e.g., things like `min_conservation` might be common?). So far we aren't in the habit of exposing all possible parameters in pipeline actions...

Second possibility (can be combined with first): don't expose these parameters directly. Parameters could be fed in through a metadata file? (akin to the good ol' qiime1 parameters file) I haven't thought through the mechanics, but thought I'd throw the idea out there.

Reply:
> akin to the good ol' qiime1 parameters file

:scream:

Reply:
I was also thinking of a parameters file, but was afraid to bring it up! :stuck_out_tongue_winking_eye:

Reply:
Perhaps we should migrate this discussion to the forum?

I think pipelines should be used to express common compositions of actions, so I don't think exposing every possible parameter is really necessary. Ideally you'd have a couple standard protocols and each could be a pipeline. Otherwise, the ordinary actions are already there, so there's nothing to stop you from making a DIY pipeline. That is actually something that I think would be worth expanding on in the future.

@turanoo I don't think you need to do anything about this in this PR.